discover/discover.go

```diff
@@ -0,0 +1,85 @@
+package discover
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/discover/aws"
+	"github.com/hashicorp/discover/azure"
+	"github.com/hashicorp/discover/gce"
+)
+
+func Discover(cfg string, l *log.Logger) ([]string, error) {
```

Comment:
I would do the map[string]string and not sure I follow on the interface argument. The go-getter style is pretty nice since you can manipulate the list of available getters and still use all the built in methods of the library. And by having a list you can append to it an init function to make a new type available. Otherwise you end up having to wrap the library.